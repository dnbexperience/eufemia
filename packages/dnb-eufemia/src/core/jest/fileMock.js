export default 'static file content'
