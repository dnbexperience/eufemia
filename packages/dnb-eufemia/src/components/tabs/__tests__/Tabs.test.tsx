/**
 * Tabs Test
 *
 */

import React from 'react'
import { axeComponent, loadScss } from '../../../core/jest/jestSetup'
import { fireEvent, render } from '@testing-library/react'
import Tabs, { TabsProps } from '../Tabs'
import Input from '../../input/Input'

const props: TabsProps = { id: 'id' }

const startup_selected_key = 'second'
const tablistData = [
  { title: 'First', key: 'first' },
  { title: 'Second', key: 'second' },
  { title: 'Third', key: 'third' },
]
const tablistDataWithContent = [
  { title: 'First', key: 1, content: <h2>First</h2> }, // without function
  { title: 'Second', key: 2, content: () => <h2>Second</h2> }, // with function
  { title: 'Third', key: 3, content: () => <h2>Third</h2> }, // with function
]
const contentWrapperData = {
  first: <h2>First</h2>, // without function
  second: () => <h2>Second</h2>, // with function
  third: <h2>Third</h2>, // without function
}

describe('Tabs component', () => {
  it('have a "selected_key" state have to be same as prop from startup', () => {
    render(
      <Tabs
        {...props}
        data={tablistData}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )
    expect(
      document
        .querySelector('.dnb-tabs__button.selected')
        .querySelectorAll('span')[0].textContent
    ).toBe(
      tablistData.find(({ key }) => key === startup_selected_key).title
    )
  })

  it('has working "on_change" and "on_click" event handler', () => {
    let preventChange = false
    const on_change = jest.fn((e) => {
      if (preventChange) {
        return false
      }
      return e
    })
    const on_click = jest.fn((e) => {
      if (preventChange) {
        return false
      }
      return e
    })

    render(
      <Tabs
        {...props}
        data={tablistData}
        on_change={on_change}
        on_click={on_click}
      >
        {contentWrapperData}
      </Tabs>
    )

    fireEvent.click(document.querySelectorAll('.dnb-tabs__button')[1])
    expect(on_change).toHaveBeenCalledTimes(1)
    expect(on_click).toHaveBeenCalledTimes(1)

    fireEvent.click(document.querySelectorAll('.dnb-tabs__button')[2])
    expect(on_change).toHaveBeenCalledTimes(2)
    expect(on_click).toHaveBeenCalledTimes(2)

    preventChange = true

    fireEvent.click(document.querySelectorAll('.dnb-tabs__button')[1])
    expect(on_change).toHaveBeenCalledTimes(2)
    expect(on_click).toHaveBeenCalledTimes(3)
  })

  it('has working "on_focus" event handler', () => {
    const on_focus = jest.fn()

    render(
      <Tabs {...props} data={tablistData} on_focus={on_focus}>
        {contentWrapperData}
      </Tabs>
    )

    fireEvent.keyDown(document.querySelector('.dnb-tabs__tabs__tablist'), {
      keyCode: 39, // right
    })
    expect(on_focus).toHaveBeenCalledTimes(1)

    fireEvent.keyDown(document.querySelector('.dnb-tabs__tabs__tablist'), {
      keyCode: 39, // right
    })
    expect(on_focus).toHaveBeenCalledTimes(2)
  })

  it('will use given tab_element', () => {
    const Link = ({ href, children }) => {
      return <a href={href}>{children}</a>
    }

    const tablistData = [
      { title: 'First', key: 'first', href: '/first' },
      { title: 'Second', key: 'second', href: '/second' },
      { title: 'Third', key: 'third', href: '/third' },
    ]

    render(
      <Tabs {...props} data={tablistData} tab_element={Link}>
        {contentWrapperData}
      </Tabs>
    )

    expect(
      document
        .querySelector('.dnb-tabs__tabs__tablist')
        .querySelectorAll('a')[1].outerHTML
    ).toMatchInlineSnapshot(
      `"<a href="/second"><span class="dnb-tabs__button__title">Second</span><span aria-hidden="true" hidden="" class="dnb-dummy">Second</span></a>"`
    )
  })

  it('should support "align" prop', () => {
    render(
      <Tabs {...props} data={tablistData} align="right">
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs__tabs')

    expect(Array.from(element.classList)).toEqual([
      'dnb-tabs__tabs',
      'dnb-tabs__tabs--right',
      'dnb-tabs__tabs--breakout',
    ])
  })

  it('should support "no_border" prop', () => {
    render(
      <Tabs {...props} data={tablistData} no_border>
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs__tabs')

    expect(Array.from(element.classList)).toEqual([
      'dnb-tabs__tabs',
      'dnb-tabs__tabs--left',
      'dnb-tabs__tabs--no-border',
      'dnb-tabs__tabs--breakout',
    ])
  })

  it('should support "content_spacing" prop', () => {
    render(
      <Tabs {...props} data={tablistData} content_spacing="small">
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs__content')

    expect(Array.from(element.classList)).toEqual(
      expect.arrayContaining([
        'dnb-tabs__content',
        'dnb-height-animation--is-in-dom',
      ])
    )
  })

  it('should support "tabs_spacing" prop', () => {
    render(
      <Tabs {...props} data={tablistData} tabs_spacing={true}>
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs__tabs')

    expect(Array.from(element.classList)).toEqual([
      'dnb-tabs__tabs',
      'dnb-tabs__tabs--left',
      'dnb-section--spacing-large',
      'dnb-tabs__tabs--breakout',
    ])
  })

  it('should support outer spacing props', () => {
    render(
      <Tabs {...props} data={tablistData} top="large">
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs')

    expect(Array.from(element.classList)).toEqual([
      'dnb-tabs',
      'dnb-space__top--large',
    ])
  })

  it('should use section component when "tabs_style" is set', () => {
    render(
      <Tabs {...props} data={tablistData} tabs_style="black-3">
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs__tabs')

    expect(element.tagName).toBe('DIV')
    expect(Array.from(element.classList)).toEqual(
      expect.arrayContaining([
        'dnb-tabs__tabs',
        'dnb-tabs__tabs--left',
        'dnb-section',
        'dnb-section--black-3',
      ])
    )
  })

  it('should use section component when "content_style" is set', () => {
    render(
      <Tabs {...props} data={tablistData} content_style="black-3">
        {contentWrapperData}
      </Tabs>
    )

    const element = document.querySelector('.dnb-tabs__content')

    expect(element.tagName).toBe('SECTION')
    expect(Array.from(element.classList)).toEqual(
      expect.arrayContaining([
        'dnb-tabs__content',
        'dnb-section',
        'dnb-section--black-3',
        'dnb-section--spacing-large',
        'dnb-no-focus',
        'dnb-space',
        'dnb-height-animation',
        'dnb-height-animation--is-in-dom',
        'dnb-height-animation--is-visible',
      ])
    )
  })

  it('border should stay inside parent boundary if breakout is set to false', () => {
    render(
      <Tabs
        {...props}
        data={tablistData}
        breakout={false}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )

    const tabs = document.querySelector('.dnb-tabs__tabs')

    expect(tabs.className).not.toContain('--breakout')
  })
})

describe('TabList component', () => {
  it('has to have the right amount of rendered components', () => {
    render(
      <Tabs
        {...props}
        data={tablistData}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )

    expect(
      document.querySelectorAll('.dnb-tabs__button__snap').length
    ).toBe(tablistData.length)
    expect(document.querySelectorAll('div[role="tabpanel"]').length).toBe(
      1
    )
  })

  it('has to have the right content on a "click event"', () => {
    render(
      <Tabs
        {...props}
        data={tablistData}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )

    fireEvent.click(document.querySelector('button[data-tab-key="third"]'))

    expect(
      document.querySelector('button[data-tab-key="third"]').classList
    ).toContain('selected')

    const content = document.querySelector('div[role="tabpanel"]')
    const { container } = render(contentWrapperData.third)
    expect(content.innerHTML).toBe(container.innerHTML)
  })
})

describe('A single Tab component', () => {
  it('has to have a role="tab" attribute and a selected class', () => {
    render(
      <Tabs
        {...props}
        data={tablistData}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )

    expect(
      document
        .querySelector('button[data-tab-key="second"]')
        .getAttribute('role')
    ).toBe('tab')
    expect(
      document.querySelector('button[data-tab-key="second"]').classList
    ).toContain('selected')
  })

  it('has to have the right content on a keydown "ArrowRight"', () => {
    render(
      <Tabs
        {...props}
        data={tablistData}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )

    // reset the state
    fireEvent.click(
      document.querySelector('button[data-tab-key="second"]')
    )

    fireEvent.keyDown(document.querySelector('div[role="tablist"]'), {
      key: 'ArrowRight',
      keyCode: 39, // right
    })
    fireEvent.click(document.querySelector('button[data-tab-key="third"]'))

    const content = document.querySelector('div[role="tabpanel"]')
    const { container } = render(contentWrapperData.third)
    expect(content.innerHTML).toBe(container.innerHTML)
  })

  it('has to work with "data only" property containing a "content"', () => {
    render(<Tabs data={tablistDataWithContent} />)
    expect(
      document
        .querySelectorAll('.dnb-tabs__button__snap')[0]
        .querySelector('button').classList
    ).toContain('selected')
    expect(
      document.querySelector('div.dnb-tabs__content').textContent
    ).toBe('First')

    // then click on tab two
    // also test the ability of having a integer only as the key
    fireEvent.click(document.querySelector('button[data-tab-key="2"]'))
    expect(
      document.querySelector('div.dnb-tabs__content').textContent
    ).toBe('Second')
  })

  it('has to run "prevent_rerender" as supposed', () => {
    render(
      <Tabs
        {...props}
        prevent_rerender
        data={[
          {
            title: 'One',
            key: 'one',
            content: () => (
              <Input label="Content one" placeholder="Edit me" />
            ),
          },
          { title: 'Two', key: 'two', content: 'Content two' },
        ]}
      />
    )

    expect(
      document.querySelector('div.dnb-tabs__cached')
    ).toBeInTheDocument()

    // also check a real live rerender scenario
    const value = 'value'
    fireEvent.change(document.querySelector('.dnb-input__input'), {
      target: { value },
    })

    // then click on tab two
    fireEvent.click(document.querySelector('button[data-tab-key="two"]'))

    // the first cache should now be hidden
    expect(
      document
        .querySelectorAll('div.dnb-tabs__cached')[0]
        .getAttribute('aria-hidden')
    ).toBe('true')

    // and on tab one again
    fireEvent.click(document.querySelector('button[data-tab-key="one"]'))

    // the entered value should still be the same
    expect(
      (document.querySelector('.dnb-input__input') as HTMLInputElement)
        .value
    ).toBe(value)

    expect(
      document
        .querySelectorAll('div.dnb-tabs__cached')[0]
        .getAttribute('aria-hidden')
    ).not.toBe('true')
    expect(
      document
        .querySelectorAll('div.dnb-tabs__cached')[1]
        .getAttribute('aria-hidden')
    ).toBe('true')
  })

  it('has to run "prerender" as supposed', () => {
    render(
      <Tabs
        {...props}
        prerender
        data={[
          {
            title: 'One',
            key: 1,
            content: 'Content one',
          },
          { title: 'Two', key: 2, content: 'Content two' },
        ]}
      />
    )

    expect(
      document.querySelector('div.dnb-tabs__cached')
    ).toBeInTheDocument()

    expect(
      document.querySelectorAll('div.dnb-tabs__cached')[0]
    ).not.toHaveAttribute('aria-hidden')
    expect(
      document
        .querySelectorAll('div.dnb-tabs__cached')[1]
        .getAttribute('aria-hidden')
    ).toBe('true')

    expect(
      document.querySelectorAll('div.dnb-tabs__cached')[0].textContent
    ).toBe('Content one')
    expect(
      document.querySelectorAll('div.dnb-tabs__cached')[1].textContent
    ).toBe('Content two')
  })

  it('has to work with "Tabs.Content" as children Components', () => {
    render(
      <Tabs {...props} data={tablistData}>
        <Tabs.Content title="first title">first</Tabs.Content>
        <Tabs.Content title="second title" selected>
          second
        </Tabs.Content>
      </Tabs>
    )
    expect(
      document
        .querySelector('button.selected')
        .getAttribute('data-tab-key')
    ).toBe('second-title')
    expect(
      document
        .querySelectorAll('.dnb-tabs__button__snap button')[1]
        .getAttribute('data-tab-key')
    ).toBe('second-title')
    expect(
      document.querySelector('div.dnb-tabs__content').textContent
    ).toBe('second')
    expect(
      document.querySelector(
        'button[aria-selected=true] span.dnb-tabs__button__title'
      ).textContent
    ).toBe('second title')
  })

  it('has to work with "Tabs.Content" from outside', () => {
    let testKey = null
    let testTitle = null
    const LinkedContent = (props: { selected_key?: string }) => {
      return (
        <>
          <Tabs id="linked" data={tablistData} {...props} />
          <Tabs.Content id="linked">
            {({ key, title }) => {
              testKey = key
              testTitle = title

              return key
            }}
          </Tabs.Content>
        </>
      )
    }

    const { rerender } = render(<LinkedContent />)

    expect(
      document
        .querySelector('button.selected')
        .getAttribute('data-tab-key')
    ).toBe('first')
    expect(testKey).toBe('first')
    expect(testTitle).toBe('First')

    rerender(<LinkedContent selected_key="second" />)

    expect(
      document
        .querySelector('button.selected')
        .getAttribute('data-tab-key')
    ).toBe('second')
    expect(testKey).toBe('second')

    fireEvent.click(document.querySelectorAll('.dnb-tabs__button')[2])

    expect(
      document
        .querySelector('button.selected')
        .getAttribute('data-tab-key')
    ).toBe('third')
    expect(testKey).toBe('third')
  })

  it('has to work with "Tabs.Content" as a single children', () => {
    render(
      <Tabs>
        <Tabs.Content title="single title">
          <div>single</div>
        </Tabs.Content>
      </Tabs>
    )

    expect(
      document.querySelector('div.dnb-tabs__content__inner').textContent
    ).toBe('single')
    expect(
      document.querySelector('button span.dnb-tabs__button__title')
        .textContent
    ).toBe('single title')
  })

  it('has to work with a single element for data property', () => {
    render(
      <Tabs
        data={[
          { title: 'single title', key: 1, content: <div>single</div> },
        ]}
      />
    )

    expect(
      document.querySelector('div.dnb-tabs__content').textContent
    ).toBe('single')
    expect(
      document.querySelector('button span.dnb-tabs__button__title')
        .textContent
    ).toBe('single title')
  })

  it('should render in StrictMode', () => {
    render(
      <React.StrictMode>
        <Tabs
          data={[
            {
              title: 'First',
              key: 'first',
            },
            {
              title: 'Second',
              key: 'second',
            },
            {
              title: 'Third',
              key: 'third',
            },
            {
              title: 'Fourth',
              key: 'fourth',
            },
          ]}
        />
      </React.StrictMode>
    )

    expect(document.querySelector('.dnb-tabs')).toBeInTheDocument()
  })

  it('should use `key` data prop if defined', () => {
    render(
      <Tabs
        data={[
          {
            title: 'First',
            key: 'first_key',
          },
          {
            title: 'Second',
            key: 'second_key',
          },
          {
            title: 'Third',
            key: 'third_key',
          },
        ]}
      />
    )
    const [first, second, third] = Array.from(
      document.querySelectorAll('.dnb-tabs__button')
    )

    expect(first).toHaveAttribute('data-tab-key', 'first_key')
    expect(second).toHaveAttribute('data-tab-key', 'second_key')
    expect(third).toHaveAttribute('data-tab-key', 'third_key')
  })
})

describe('Tabs scss', () => {
  it('has to match style dependencies css', () => {
    const css = loadScss(require.resolve('../style/deps.scss'))
    expect(css).toMatchSnapshot()
  })

  it('have to match default theme snapshot', () => {
    const css = loadScss(
      require.resolve('../style/themes/dnb-tabs-theme-ui.scss')
    )
    expect(css).toMatchSnapshot()
  })
})

describe('Tabs ARIA', () => {
  it('should validate with ARIA rules', async () => {
    const Comp = render(
      <Tabs
        {...props}
        data={tablistData}
        selected_key={startup_selected_key}
      >
        {contentWrapperData}
      </Tabs>
    )
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})
