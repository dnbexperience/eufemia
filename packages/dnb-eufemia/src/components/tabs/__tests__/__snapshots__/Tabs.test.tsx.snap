// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs scss has to match style dependencies css 1`] = `
"/*
* Used for snapshot testing
*
*/
/*
 * Utilities
 */
/*
 * Tabs component
 *
 */
/*
 * Utilities
 */
/*
* Button mixins
*
*/
.dnb-tabs {
  --tablist-border-color--default: grey;
  --tab-title-font-weight: var(--font-weight-regular);
  --tab-title-font-size: var(--font-size-basis);
  --tab-title-font-color: dimgray;
  --tab-title-background--hover: darkgrey;
  --tab-title-color--hover: inherit;
  --tab-title-border--hover: black;
  --tab-title-background--disabled: inherit;
  --tab-title-color--disabled: lightgrey;
  --tab-title-border--active-selected: black;
  --tab-title-color--active-selected: black;
  --tab-title-font-weight--selected: var(--font-weight-medium);
  /* stylelint-disable */
  /* stylelint-enable */
}
.dnb-tabs__tabs {
  position: relative;
  z-index: 1;
  display: flex;
  align-items: center;
}
.dnb-tabs__tabs, .dnb-tabs__tabs.dnb-section--spacing {
  padding-bottom: 0;
}
.dnb-tabs__tabs__tablist {
  display: flex;
  flex: 0 1 auto;
  overflow-x: auto;
  overscroll-behavior: contain;
  -webkit-overflow-scrolling: touch;
  -ms-overflow-style: auto;
  scrollbar-color: var(--scrollbar-thumb-color, #888) transparent;
  /* Hide scrollbar for Chrome, Safari */
  /* stylelint-disable-next-line */
  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */
}
.dnb-tabs__tabs__tablist:focus {
  outline: none;
  border-radius: 0.5rem;
}
html[data-whatinput=keyboard] .dnb-tabs__tabs__tablist:focus {
  --border-color: var(--focus-ring-color);
  --border-width: var(--focus-ring-width);
  box-shadow: 0 0 0 var(--border-width) var(--border-color);
  border-color: transparent;
}
html:not([data-visual-test]) .dnb-tabs__tabs__tablist {
  scroll-behavior: smooth;
}
@supports not (scrollbar-color: auto) {
  .dnb-tabs__tabs__tablist::-webkit-scrollbar {
    border-radius: var(--scrollbar-thumb-width, 0.5rem);
    background-color: var(--scrollbar-track-color, #eee);
  }
  .dnb-tabs__tabs__tablist::-webkit-scrollbar:vertical {
    width: var(--scrollbar-track-width, 0.5rem);
  }
  .dnb-tabs__tabs__tablist::-webkit-scrollbar:horizontal {
    height: var(--scrollbar-track-width, 0.5rem);
  }
  .dnb-tabs__tabs__tablist::-webkit-scrollbar-thumb {
    background-color: var(--scrollbar-thumb-color, #888);
    border-radius: var(--scrollbar-thumb-width, 0.5rem);
  }
  .dnb-tabs__tabs__tablist::-webkit-scrollbar-thumb:hover {
    background-color: var(--scrollbar-thumb-hover-color, #666);
  }
}
.dnb-tabs__tabs__tablist::-webkit-scrollbar {
  display: none;
}
.dnb-tabs__tabs--left .dnb-tabs__tabs__tablist {
  justify-content: flex-start;
}
.dnb-tabs__tabs--right .dnb-tabs__tabs__tablist {
  flex: 1;
  justify-content: flex-end;
}
.dnb-tabs__tabs--center .dnb-tabs__tabs__tablist {
  justify-content: center;
}
.dnb-tabs__tabs::before {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: 0;
  width: 100%;
  height: 1px;
}
.dnb-tabs__tabs--no-border::before {
  content: none;
}
.dnb-tabs__tabs--breakout::before {
  left: -100vw;
  width: 100vw;
}
.dnb-tabs__tabs:not(.dnb-section)::before {
  background-color: var(--tablist-border-color--default);
  box-shadow: 100vw 0 0 0 var(--tablist-border-color--default);
}
.dnb-tabs__scroll-nav-button {
  position: absolute;
  z-index: 2;
  right: -2px;
  display: none;
  opacity: 0;
  pointer-events: none;
  transition: opacity 600ms ease-out;
}
.dnb-tabs__scroll-nav-button:first-of-type {
  left: -2px;
  right: auto;
}
.dnb-tabs__scroll-nav-button--visible {
  display: flex;
  pointer-events: initial;
  opacity: 1;
}
.dnb-tabs__scroll-nav-button--hide {
  pointer-events: none;
  opacity: 0 !important;
}
.dnb-tabs--at-edge .dnb-tabs__scroll-nav-button:first-of-type {
  border-radius: 0 50% 50% 0;
}
.dnb-tabs--at-edge .dnb-tabs__scroll-nav-button:last-of-type {
  border-radius: 50% 0 0 50%;
}
.dnb-tabs--at-edge .dnb-tabs__tabs__tablist:focus {
  border-radius: 0;
}
.dnb-tabs__button, .dnb-core-style .dnb-tabs .dnb-tabs__button {
  position: relative;
  z-index: 1;
  user-select: none;
  /* stylelint-disable-next-line */
  -webkit-user-select: none;
  margin: 0;
  padding: 0 0.125rem; /* 1/16*4 */
  background-color: transparent;
  border-bottom: 1px solid transparent;
  cursor: pointer;
  outline: none;
  border: none;
  font-size: var(--font-size-basis);
  line-height: var(--line-height-basis);
  text-decoration: none;
  white-space: nowrap;
  transition: margin 400ms ease-out;
  /* stylelint-disable no-descending-specificity */
  /* stylelint-enable no-descending-specificity */
}
.dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button__title {
  font-weight: var(--tab-title-font-weight);
  font-size: var(--tab-title-font-size);
  text-align: center;
  display: block;
  color: var(--tab-title-font-color);
  padding: 0.5rem 0 0.25rem;
}
html:not([data-whatintent=touch]) .dnb-tabs__button__title:hover[disabled], html:not([data-whatintent=touch]) .dnb-core-style .dnb-tabs .dnb-tabs__button__title:hover[disabled] {
  cursor: not-allowed;
}
html:not([data-whatintent=touch]) .dnb-tabs__button__title:hover:not([disabled]), html:not([data-whatintent=touch]) .dnb-core-style .dnb-tabs .dnb-tabs__button__title:hover:not([disabled]) {
  color: var(--tab-title-color--hover);
}
.dnb-tabs__button__title .dnb-icon, .dnb-core-style .dnb-tabs .dnb-tabs__button__title .dnb-icon {
  font-size: var(--font-size-small);
  transform: translateY(-0.125rem);
}
html:not([data-whatintent=touch]) .dnb-tabs__button:hover[disabled], html:not([data-whatintent=touch]) .dnb-core-style .dnb-tabs .dnb-tabs__button:hover[disabled] {
  cursor: not-allowed;
}
html:not([data-whatintent=touch]) .dnb-tabs__button:hover:not([disabled]), html:not([data-whatintent=touch]) .dnb-core-style .dnb-tabs .dnb-tabs__button:hover:not([disabled]) {
  background-color: var(--tab-title-background--hover);
}
html:not([data-whatintent=touch]) .dnb-tabs__button:hover:not([disabled])::after, html:not([data-whatintent=touch]) .dnb-core-style .dnb-tabs .dnb-tabs__button:hover:not([disabled])::after {
  height: 2px;
  border-radius: 2px;
  background-color: var(--tab-title-border--hover);
}
.dnb-tabs__button .dnb-dummy, .dnb-core-style .dnb-tabs .dnb-tabs__button .dnb-dummy {
  display: flex;
  flex-direction: column;
  height: 0;
  visibility: hidden;
  overflow: hidden;
}
.dnb-tabs__button .dnb-dummy, .dnb-core-style .dnb-tabs .dnb-tabs__button .dnb-dummy {
  margin-top: 0.25rem;
}
.dnb-tabs__button::after, .dnb-core-style .dnb-tabs .dnb-tabs__button::after {
  content: "";
  position: absolute;
  z-index: -1;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 1px;
}
.dnb-tabs__button[disabled], .dnb-core-style .dnb-tabs .dnb-tabs__button[disabled] {
  background-color: var(--tab-title-background--disabled);
}
.dnb-tabs__button[disabled] .dnb-tabs__button__title, .dnb-tabs__button[disabled] .dnb-core-style .dnb-tabs .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button[disabled] .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button[disabled] .dnb-core-style .dnb-tabs .dnb-tabs__button__title {
  color: var(--tab-title-color--disabled);
}
.dnb-tabs__button span[hidden], .dnb-core-style .dnb-tabs .dnb-tabs__button span[hidden] {
  font-weight: var(--tab-title-font-weight--selected);
}
.dnb-tabs__button:focus, .dnb-core-style .dnb-tabs .dnb-tabs__button:focus {
  overflow: visible;
}
.dnb-tabs__button:focus::before, .dnb-core-style .dnb-tabs .dnb-tabs__button:focus::before {
  content: "";
  position: absolute;
  z-index: 1;
  top: 0;
  left: -0.5rem;
  bottom: 0;
  right: -0.5rem;
  height: inherit;
  border-radius: inherit;
  outline: none;
}
html[data-whatinput=keyboard] .dnb-tabs__button:focus::before, html[data-whatinput=keyboard] .dnb-core-style .dnb-tabs .dnb-tabs__button:focus::before {
  --border-color: var(--focus-ring-color);
  --border-width: var(--focus-ring-width);
  box-shadow: 0 0 0 var(--border-width) var(--border-color);
  border-color: transparent;
}
.dnb-tabs__button:focus::before, .dnb-core-style .dnb-tabs .dnb-tabs__button:focus::before {
  top: 0.5rem;
  left: -0.375rem;
  right: -0.375rem;
  bottom: 0.5rem;
  border-radius: 1.5rem;
  background-color: var(--tab-focus-background-color, none);
}
.dnb-tabs__button:not([disabled]):active::after, .dnb-tabs__button:not([disabled]).selected::after, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]):active::after, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]).selected::after {
  height: 2px;
  border-radius: 2px;
  background-color: var(--tab-title-border--active-selected);
}
.dnb-tabs__button:not([disabled]):active .dnb-tabs__button__title, .dnb-tabs__button:not([disabled]).selected .dnb-tabs__button__title, .dnb-tabs__button:not([disabled]):active .dnb-core-style .dnb-tabs .dnb-tabs__button__title, .dnb-tabs__button:not([disabled]).selected .dnb-core-style .dnb-tabs .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]):active .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]).selected .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]):active .dnb-core-style .dnb-tabs .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]).selected .dnb-core-style .dnb-tabs .dnb-tabs__button__title {
  color: var(--tab-title-color--active-selected);
}
.dnb-tabs__button:not([disabled]).selected .dnb-tabs__button__title, .dnb-tabs__button:not([disabled]).selected .dnb-core-style .dnb-tabs .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]).selected .dnb-tabs__button__title, .dnb-core-style .dnb-tabs .dnb-tabs__button:not([disabled]).selected .dnb-core-style .dnb-tabs .dnb-tabs__button__title {
  font-weight: var(--tab-title-font-weight--selected);
}
.dnb-tabs__button__snap {
  display: flex;
  padding: 0 1rem 0 1.5rem;
  will-change: padding;
  transition: padding 1s var(--easing-default);
}
@media screen and (max-width: 40em) {
  .dnb-tabs__button__snap {
    padding: 0 1rem;
  }
}
.dnb-tabs__button__snap:first-of-type {
  padding-left: 0;
}
.dnb-tabs__button__snap:last-of-type {
  padding-right: 0.5rem;
}
html[data-whatinput=keyboard] .dnb-tabs__button__snap:first-of-type.focus .dnb-tabs__button:focus {
  margin-left: 0.5rem;
}
html[data-whatinput=keyboard] .dnb-tabs__button__snap:last-of-type.focus {
  /* stylelint-disable */
  /* stylelint-enable */
}
html[data-whatinput=keyboard] .dnb-tabs__button__snap:last-of-type.focus .dnb-tabs__button:focus {
  margin-right: 0.5rem;
}
.dnb-tabs__cached {
  opacity: 1;
  transition: opacity 1s var(--easing-default);
}
.dnb-tabs__cached--hidden {
  opacity: 0;
  visibility: hidden;
  height: 0 !important;
  overflow: hidden;
}
html[data-visual-test] .dnb-tabs .dnb-tabs__cached, html[data-visual-test] .dnb-tabs .dnb-tabs__button, html[data-visual-test] .dnb-tabs .dnb-tabs__button__snap, html[data-visual-test] .dnb-tabs .dnb-tabs__scroll-nav-button {
  transition: none !important;
}
.dnb-tabs__content {
  padding-bottom: 0;
}
.dnb-tabs__content:focus {
  position: relative;
}
.dnb-tabs__content:focus::before {
  content: "";
  position: absolute;
  z-index: -1;
  top: 0.125rem;
  bottom: -0.5rem;
  left: -0.5rem;
  right: -0.5rem;
  max-width: 100vw;
  border-radius: 0.5rem;
  outline: none;
}
html[data-whatinput=keyboard] .dnb-tabs__content:focus::before {
  --border-color: var(--focus-ring-color);
  --border-width: var(--focus-ring-width);
  box-shadow: 0 0 0 var(--border-width) var(--border-color);
  border-color: transparent;
}"
`;

exports[`Tabs scss have to match default theme snapshot 1`] = `
"/*
* Tabs theme
*
*/
/*
 * Utilities
 */
/*
* Button mixins
*
*/
.dnb-tabs {
  --tablist-border-color--default: var(--color-black-8);
  --tab-title-font-color: var(--color-sea-green);
  --tab-title-font-weight: var(--font-weight-regular);
  --tab-title-font-size: var(--font-size-basis);
  --tab-title-background--hover: var(--color-mint-green-50);
  --tab-title-color--hover: var(--color-sea-green);
  --tab-title-border--hover: var(--color-sea-green);
  --tab-title-color--disabled: var(--color-sea-green-30);
  --tab-title-border--active-selected: var(--color-sea-green);
  --tab-title-color--active-selected: var(--color-emerald-green);
}
.dnb-tabs__button__title {
  font-weight: var(--font-weight-regular);
  font-size: var(--font-size-basis);
  text-align: center;
  display: block;
  color: var(--color-sea-green);
  padding: 0.5rem 0 0.25rem;
}
html:not([data-whatintent=touch]) .dnb-tabs__button:hover[disabled] {
  cursor: not-allowed;
}
html:not([data-whatintent=touch]) .dnb-tabs__button:hover:not([disabled]) {
  background-color: var(--color-mint-green-50);
}
html:not([data-whatintent=touch]) .dnb-tabs__button:hover:not([disabled])::after {
  height: 2px;
  border-radius: 2px;
  background-color: var(--color-sea-green);
}
.dnb-tabs__button:focus {
  overflow: visible;
}
.dnb-tabs__button:focus::before {
  content: "";
  position: absolute;
  z-index: 1;
  top: 0;
  left: -0.5rem;
  bottom: 0;
  right: -0.5rem;
  height: inherit;
  border-radius: inherit;
  outline: none;
}
html[data-whatinput=keyboard] .dnb-tabs__button:focus::before {
  --border-color: var(--focus-ring-color);
  --border-width: var(--focus-ring-width);
  box-shadow: 0 0 0 var(--border-width) var(--border-color);
  border-color: transparent;
}
.dnb-tabs__button:focus::before {
  top: 0.5rem;
  left: -0.375rem;
  right: -0.375rem;
  bottom: 0.5rem;
  border-radius: 1.5rem;
}
.dnb-tabs__button[disabled] .dnb-tabs__button__title {
  color: var(--color-sea-green-30);
}
.dnb-tabs__button:not([disabled]):active::after, .dnb-tabs__button:not([disabled]).selected::after {
  height: 2px;
  border-radius: 2px;
  background-color: var(--color-sea-green);
}
.dnb-tabs__button:not([disabled]):active .dnb-tabs__button__title, .dnb-tabs__button:not([disabled]).selected .dnb-tabs__button__title {
  color: var(--color-emerald-green);
}
.dnb-tabs__button:not([disabled]).selected .dnb-tabs__button__title {
  font-weight: var(--font-weight-medium);
}
.dnb-tabs__button span[hidden] {
  font-weight: var(--font-weight-medium);
}"
`;
